core/types/calls.cc

```diff
@@ -2919,6 +2937,81 @@ class Magic_mergeHashValues : public IntrinsicMethod {
     }
 } Magic_mergeHashValues;
 
+namespace {
+void convertDefinedToType(const GlobalState &gs, const DispatchArgs &args, DispatchResult &res) {
+    // args[0] is the receiver
+    // args[1] is the symbol name of the variable (class or instance)
+    if (args.args.size() != 2) {
+        return;
+    }
+
+    auto &receiver = args.args[0];
+    if (receiver->type.isUntyped()) {
+        res.returnType = receiver->type;
+        return;
+    }
+    if (!receiver->type.isFullyDefined()) {
```

Comment:
Do you have a test for this?

Reply:
I do not.  This was cargo-culted from the `<Magic>.<call-with-*>` intrinsics.  I'm not entirely sure how one would write a test for this, and it seems reasonable to give up on such types in any event?

Reply:
```ruby
# typed: true

class A
  extend T::Generic

  Elem = type_member

  @@x = T.let(nil, T.nilable(Elem))

  sig {params(x: Elem).void}
  def foo(x)
    @@x ||= x
  end
end
```

Your PR crashes with a backtrace on this test case.